### ISSUE TITLE
Bit field of enum type causes crash when generating DxilFieldAnnotation

### ISSUE THREAD

**Issue by tex3d** [microsoft/DirectXShaderCompiler#5257]
The following test produces a crash during clang codegen.

```hlsl
// RUN: %dxc -T lib_6_6 -HV 2021 %s

enum SomeEnum { Val1 };

struct SomeStruct
{
    // For some reason a uint32_t starting element is needed to allow for conversion from literal 0 in Sema
    uint32_t m1 : 16;
    SomeEnum m3 : 3;
};

export int SomeFunc()
{
    SomeStruct s = (SomeStruct)0;
    s.m3 = Val1;
    return (int)s.m3;
}
```

This code in `CGHLSLMS.cpp` assumes the bitfield type is a `clang::BuiltinType`, which is not true when it's an enum.
https://github.com/microsoft/DirectXShaderCompiler/blob/19c32587bee40189e8805f6dc12d96adb3550e2e/tools/clang/lib/CodeGen/CGHLSLMS.cpp#L1012

**Comment by tex3d** [comment]
@python3kgae note, this crash would have been introduced from this PR: #5142

Before it wouldn't have had reflection for the bitfields, but it also may not have crashed when using an enum in a bitfield.

**Comment by pow2clk** [comment]
Fixed by #5260
